src/containers/WelcomePage/FrontpageSubjects.jsx

```diff
@@ -136,41 +152,69 @@ const LinkToAbout = () => (
   </a>
 );
 
-class FrontpageSubjects extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { preview: false };
-  }
-
-  componentDidMount() {
+const FrontpageSubjects = ({ categories, subjects, locale }) => {
+  const [preview, setPreview] = useState(false);
+  const [showImported, setShowImported] = useState(false);
+  useEffect(() => {
     if (
       document.location.search &&
       document.location.search.includes('preview')
     ) {
-      this.setState({ preview: true });
+      setPreview(true);
     }
-  }
-
-  render() {
-    const { categories, subjects, locale } = this.props;
-    const frontpageCategories = mapHardCodedCategories(
-      categories,
-      locale,
-      this.state.preview,
-    );
-
-    const allSubjects = config.isNdlaProdEnvironment
-      ? frontpageCategories
-      : [...frontpageCategories, getAllImportSubjectsCategory(subjects)];
+  });
 
-    return (
-      <FrontpageSubjectsSection
+  const frontpageCategories = mapHardCodedCategories(
+    categories,
+    locale,
+    preview,
+  );
+
+  const allSubjects = config.isNdlaProdEnvironment
+    ? frontpageCategories
+    : [...frontpageCategories, getAllImportSubjectsCategory(subjects)];
+
+  const allSubjectsWithFIxedEndSlash = allSubjects.map(category => ({
+    ...category,
+    subjects: category.subjects
+      ? category.subjects.map(subject => ({
+          ...subject,
+          url: subject && subject.url ? fixEndSlash(subject.url) : subject.url,
+        }))
+      : [],
+  }));
+  const imported = allSubjectsWithFIxedEndSlash.find(
```

Comment:
Fixed og ikke FIxed. 

I tillegg tenker jeg kanskje at det skjer ganske mye her? Er det muligheter for å få laget noen tester på dette?

Reply:
jepp 👍